rust/template/distributed_datalog/src/sinks/file.rs

```diff
@@ -0,0 +1,171 @@
+use std::fmt::Debug;
+use std::fmt::Display;
+use std::fs::File as FsFile;
+use std::fs::OpenOptions;
+use std::io::Error;
+use std::marker::PhantomData;
+use std::path::Path;
+
+use differential_datalog::program::Update;
+use differential_datalog::record_val_upds;
+use differential_datalog::DDlogConvert;
+use differential_datalog::RecordReplay;
+
+use crate::Observer;
+
+/// An object implementing the `Observer` interface and dumping
+/// transaction traces into a file.
+#[derive(Debug)]
+pub struct File<C> {
+    /// The file we dump our transaction traces into.
+    file: FsFile,
+    /// Unused phantom data.
+    _unused: PhantomData<C>,
+}
+
+impl<C> File<C> {
+    /// Create a new file based observer using the given file.
+    pub fn new(file: FsFile) -> Self {
+        Self {
+            file,
+            _unused: PhantomData,
+        }
+    }
+
+    /// Create a new file based observer using a file at the given path
+    /// that will be opened and appended to (if it exists) or newly
+    /// created.
+    pub fn open<P>(path: P) -> Result<Self, Error>
+    where
+        P: AsRef<Path>,
+    {
+        Ok(Self {
+            file: OpenOptions::new()
+                .create(true)
+                .write(true)
+                .append(true)
```

Comment:
It might be good to have a flag in the `new()` method to choose whether to set `append` to `true`.

Reply:
Let me remove the method for now, so we don't have any such policy stuff in there to begin with.